sparse/coo/core.py

```diff
@@ -1704,6 +1704,9 @@ def reshape(self, shape, order='C'):
 
         if self.shape == shape:
             return self
+        
+        if np.prod(self.shape) != np.prod(shape):
```

Comment:
Should either be changed to `reduce` as in the shape or `dtype=np.intp` added.

Reply:
This looks like remnants of my past PR. I'm not sure how but this commit should be deleted.

Reply:
It's fine, just don't delete it. Just rewrite the line in a subsequent PR.